aws-api/src/androidTest/java/com/amplifyframework/api/aws/CodeGenerationInstrumentationTest.java

```diff
@@ -118,6 +123,12 @@ public void subscribeReceivesMutationEvent() {
         SynchronousApi.Subscription<Person> subscription =
             api.onCreate(PERSON_API_NAME, Person.class);
 
+        // Give the subscription time to complete the connection before running the mutation.
+        // This should happen within 1 second - if not, it's good that this fails so we can investigate.
+        try {
+            Thread.sleep(ONE_SECOND_OF_MILLISECONDS);
+        } catch (InterruptedException exception) { }
```

Comment:
Ah crumbs. I think this is why I had left this part of the method as blocking. If the subscription call returns before getting acknowledgement, we have this new issue - how to do we convey the acknowledgement to the user? It used to be synonymous with getting a subscription ID back.

The user would also need to sleep for a random amount of time, like this test does.

If that part is now async, we can provide the information with a callback. Like:
```
/**
 * Subscribe to mutations that occur for a given model class.
 * @param clazz Type of model being subscribed
 * @param subscriptionType type of events to subscribe to
 * @param onSubscriptionStarted Called when a subscription begins; a subscription ID is passed
 * @param onNextItem Called when a new subscription item is available
 * @param onSubscriptionFailed Called when the subscription terminates due to a failure
 * @param onSubscriptionCompleted Called when a subscription terminates gracefully
 */
<T extends Model> void subscribe(
    Class<T> clazz,
    SubscriptionType subscriptionType,
    Consumer<String> onSubscriptionStarted,
    Consumer<T> onNextItem,
    Consumer<ApiException> onSubscriptionFailed,
    Action onSubscriptionCompleted
);
```

Am (irrelevant) note on mechanics, here -- you can use the [`Sleep#milliseconds(long)`](https://github.com/aws-amplify/amplify-android/blob/master/testutils/src/main/java/com/amplifyframework/testutils/Sleep.java#L30) test utility to clean things up a bit and get rid of this exception block.

Reply:
Updated PR to do this